core/src/main/scala/org/bykn/bosatsu/rankn/DefinedType.scala

```diff
@@ -22,15 +20,15 @@ final case class DefinedType[+A](
   /**
    * A type with exactly one constructor is a struct
    */
-  def isStruct: Boolean = constructors.lengthCompare(1) == 0
+  val isStruct: Boolean = constructors.lengthCompare(1) == 0
 
   /**
    * A newtype is just a wrapper for another type.
    * It could be removed statically from the program
    */
-  def isNewType: Boolean =
+  val isNewType: Boolean =
```

Comment:
I wonder if it might be nice to make this an `Option[ConstructorFn]`? So if `constructors` has the properties you want then it'll be returned for you to use.

Reply:
I'm going to punt on this. I think what we really want is something like:

```scala
def asNewType: Option[NewTypeConstructorFn]
```
which has exactly one constructor function. To do that without duplication you might have:
```scala
case class ConstructorFn[F[_]](name: Constructor, args: F[(Bindable, Type)], fnType: Type)
```
Then we can have `ConstructorFn[List]` for the generic case or `ConstructorFn[cats.Id]` for the new-type case....

But at the end of the day, you are guarding against a small set of bugs that we already have tests for and complicating all uses of the code. So, I think the cost/benefit trade-off is a bit weak at the moment.